### ISSUE TITLE
How to code a new benchmark suite? A second API? 

### ISSUE THREAD

**Issue by GoogleCodeExporter** [numbbo/coco#92]
```
Do we have a recipe/guidance how to code a new benchmark suite? That would be 
awesome, also in order to get comparatively quick insight into the inner 
workings of the lib.  

```

Original issue reported on code.google.com by `nikolaus...@gmail.com` on 26 Feb 2015 at 4:48
* Blocking: #93, #94

**Comment by ttusar** [comment]
I have a question regarding suites (I couldn't figure this out from the code at its current state): how will the user interface COCO problems/suites? 
One possibility is to iterate through problem indices using coco_suite_get_next_problem_index, but the two demos (the single- and multi-objective ones) seem to also allow calls to encode/decode problem indices. Was this second variant only temporary or is it supposed to be supported? I'm asking because while we do have some functions to encode/decode problem indices, they are currently not visible to the user.

**Comment by nikohansen** [comment]
The original C implementation has indeed hidden the indexed access to single functions of a suite. I believe the reason is to "encourage" producing results for the entire suite only. I re-implemented random access in Python. So to me it seems quite often useful to have.

**Comment by brockho** [comment]
depending on #162

**Comment by ttusar** [comment]
Closed by commit a3bc14288a5adfa5dbd3b4ea76a01ce5f9a3c41 to the development branch.

**Comment by nikohansen** [comment]
_Second API_ meant recipe/guidance how to code a new benchmark suite/testbed, that is, an API for constructing a new testbed rather than for running an existing testbed. It might be this issue was closed prematurely, or do we have a recipe for this?

**Comment by ttusar** [comment]
We will have some text about this in the coco-doc/C readme (for a later release).
Is this enough?

**Comment by nikohansen** [comment]
Yes, that's probably sufficient. In general however, to separate docs from code is suboptimal for maximizing the chance to keeping the docs up-to-date with the code.

**Comment by brockho** [comment]
If I understand @ttusar's latest comment on this issue, code and documentation are/will be together (in the C code) and then exported via doxygen. Can this issue therefore be closed, @ttusar or do we need to check/update the documentation?

**Comment by ttusar** [comment]
We were talking about [this](http://numbbo.github.io/coco-doc/C/) documentation, which is not included in the C code. It is also still missing two items (How to write new test functions and combine them into test suites and How to write additional performance indicators and logging functionality). 

I believe we came to the conclusion that the separation of the documentation from the code is not optimal, but will have to do for the moment. The issue should remain open to remind us of the missing text (or maybe we could close it and open a new one to make it clearer what it is about).

**Comment by ttusar** [comment]
I will try to update this documentation soon, but in the mean time, we still have a dead link in the [documentation of the C code](http://numbbo.github.io/coco-doc/C/):

> How to conduct experiments in all supported languages is described at http://numbbo.github.io/bbob-biobj-experiments-doc (coming soon)

I don't remember what was/is the plan for this documentation... What should I do with this text?

**Comment by nikohansen** [comment]
superseded by the generic http://numbbo.github.io/coco-doc/experimental-setup/ ?

**Comment by ttusar** [comment]
I have updated the documentation, please have a look and make any necessary changes. 
I think this issue can be closed now.
